### ISSUE TITLE
Fix `at-rule-no-unknown` false positive for `@starting-style`

### ISSUE THREAD

**Issue by Mouvedia** [stylelint/stylelint#7420]
### What minimal example or steps are needed to reproduce the bug?

```css
@starting-style {}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "at-rule-no-unknown": true
  }
}
```

### How did you run Stylelint?

https://stylelint.io/demo

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "latest",
}
```

### What did you expect to happen?

`@starting-style` to be _known_ since it has a decent browser support.
see https://caniuse.com/mdn-css_at-rules_starting-style

### What actually happened?

Unexpected unknown at-rule "@starting-style"

### Do you have a proposal to fix the bug?

Add `starting-style` to
https://github.com/stylelint/stylelint/blob/b92260f415ae2e1c1ee053ea01658be0b063899b/lib/reference/atKeywords.mjs#L4
and add `nestingSupportedAtKeywords` to
https://github.com/stylelint/stylelint/blob/b92260f415ae2e1c1ee053ea01658be0b063899b/lib/reference/atKeywords.mjs#L33

**Comment by ybiquitous** [comment]
@Mouvedia Thanks for the proposal with a reproducible demo. Sounds good. I didn't know that new at-rule. 👍🏼 

I've labeled the issue as **ready to implement**. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

---

Incidentally, sharing a reproducible demo link would be more helpful. For example:
https://stylelint.io/demo/#N4Igxg9gJgpiBcIACBnALgQwE5oJYDsBzAWnQE8AbGAAmAF8QAaEAM1yoDkMBbORGAB48ADlQB0YFCibgI+NoQQhgAHXzVqKkFgCuVFFvi01GjVoxpiuqsXwRiO-AGs7Ad3yHqaXTBPU6agzMkPK4hABiEFjcFkoAVihyMrDC0oiq6pog5FQUBGieWgCMAGxiAAxiAExagSB0QA

**Comment by romainmenke** [comment]
`@starting-style` should also be added to the list of at rules that are allowed in nesting.

https://github.com/stylelint/stylelint/blob/main/lib/reference/atKeywords.mjs#L4-L10

https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0AzCEAEwA6A7DGkANhAE7wa4QAuAtAIa3FkDceeBAAgM7X2nUAlrgDmtXgE8iMLBwIYA9ACoMMAB4AHGGGowoGevhilSZDAAtjMpQrkFm5SjQZMIJUm3wYAvnm8gAGhAUQWkAOXoAWzhEdSiNaSQwbm5A8AhcEJEEEBwvbBBSAFdpbgKKPPkMAtgwIn56IQzaDVMtAQlaADd6IiKYWipaItwAayoAd1xyjGpimDlfXH8gyEzBEQAxMkjGnIArbgy02A1UxErqkElpImFqGYKARgA2JAAGJAAmAr8QbyAA

**Comment by romainmenke** [comment]
On re-reading this issue I noticed that the proposed change here is to also add `nestingSupportedAtKeywords` to `atKeywords`. This was previously discussed during review and the current form was what was chosen.

Being able to be nested is an attribute that any at rule can have.
It is only a single dimension.

Future additions to CSS can introduce other dimensions.
At rules can be part of one or more sub groups and being able to be nested is just one of these.

------

`pageMarginAtKeywords` however are different. This is a fixed group of at rules and they are defined by being the "page margin at rules". e.g. `@media` can never become a page margin at rule in the future.

------

I personally prefer the explicit notation :

```
const groupingOfAtRulesOne = new Set(['foo', 'bar']);
const groupingOfAtRulesTwo = new Set(['bar', 'baz']);

const allAtRules = new Set(['foo', 'bar', 'baz', 'other']);
```

Other option 1:

```
const groupingOfAtRulesOne = new Set(['foo', 'bar']);
const groupingOfAtRulesTwo = new Set(['bar', 'baz']);

const allAtRules = uniteSets(groupingOfAtRulesOne, groupingOfAtRulesTwo, ['foo', 'bar', 'baz', 'other']);
```

Other option 2:

```
const groupingOfAtRulesOne = new Set(['foo', 'bar']);
const groupingOfAtRulesTwo = new Set(['bar', 'baz']);

const allAtRules = uniteSets(groupingOfAtRulesOne, groupingOfAtRulesTwo, ['other']);
```

I think it's best to decide on something and stick with that unless there is a very good reason to change it. I don't have a strong opinion (anymore) on what is better here.
I do prefer consistency so that we have fewer changes unrelated to the actual issue.

**Comment by ybiquitous** [comment]
@romainmenke Thanks for digging into this issue. As you suggested, minimizing the scope of changes seems to make sense here. So, I agree with adding `@starting-style` to `atKeywords` rather than `nestingSupportedAtKeywords`. The addition will affect only on `at-rule-no-unknown`:

https://github.com/stylelint/stylelint/blob/19ab06a4ca55da15ff64a6391037a47a01136229/lib/rules/at-rule-no-unknown/index.mjs#L2

**Comment by Mouvedia** [comment]
`@starting-style` can be nested.
see https://drafts.csswg.org/css-transitions-2/#example-8d9992e9

`nestingSupportedAtKeywords` will always be part of `atKeywords`.
This can be discussed in the PR; it's only an implementation detail.

**Comment by romainmenke** [comment]
Yeah, content-wise @Mouvedia is fully correct :)
It can be nested.

My comment was purely about the implementation and how different ways to structure the data reflect the content and the properties of that content. Any change is fine by me.

**Comment by ybiquitous** [comment]
I see. So, adding `@starting-style` to `nestingSupportedAtKeywords` would fix also false positives in `declaration-property-value-no-unknown`, right?

https://github.com/stylelint/stylelint/blob/19ab06a4ca55da15ff64a6391037a47a01136229/lib/rules/declaration-property-value-no-unknown/index.mjs#L164

**Comment by romainmenke** [comment]
Yes :)
